### PR TITLE
docs: clarify `promise-function-async` docs

### PR DESCRIPTION
**Pull request by benmccann**
## PR Checklist

- [ ] Addresses an existing open issue - no issue, but a maintainer also suggested these docs could be updated in https://github.com/typescript-eslint/typescript-eslint/issues/9284#issuecomment-2150757249
- [ ] That issue was marked as [accepting prs](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aopen+is%3Aissue+label%3A%22accepting+prs%22)
- [ ] Steps in [Contributing](https://typescript-eslint.io/contributing) were taken - N/A

## Overview

Reword the description much more simply

Fix an ungrammatical phrase in the callout that was hard to parse - "make the return type explicitly to allow the rule to pass"

### CODE REVIEW

**Review by StyleShit — APPROVED** [review]
Sounds much better IMO 😄

File: packages/eslint-plugin/docs/rules/promise-function-async.mdx [hunk 1]
```diff
@@ -9,16 +9,9 @@ import TabItem from '@theme/TabItem';
 >
 > See **https://typescript-eslint.io/rules/promise-function-async** for documentation.
 
-Ensures that each function is only capable of:
+Ensures that functions returning promises can only return rejected promises when an error is encountered and cannot throw. This simplifies the API for callers by making it so that they only need to handle errors in a single manner.
```
Comment (JoshuaKGoldberg) [inline packages/eslint-plugin/docs/rules/promise-function-async.mdx:12]: 🤔 I don't think this is better at explaining the rule. At its core, `promise-function-async` enforces that _"any function or method that returns a Promise to be marked async"_. This explanation is a rewording of the technical backing but both are pretty in-the-weeds.

I think I'd rather take this on myself as part of #8088 actually. Unless @kirkwaiblinger do you have specific wording you wanted to work with?

Comment (kirkwaiblinger) [reply packages/eslint-plugin/docs/rules/promise-function-async.mdx:12]: I agree that this is in need of more significant overhaul from a team member 👍

**Comment by kirkwaiblinger** [comment]
Hey @benmccann! 👋 

We appreciate the suggestions and grammar fixup! I'm going to close this PR in light of https://github.com/typescript-eslint/typescript-eslint/pull/9289#discussion_r1643314131, but, to be clear, that doesn't indicate that there's anything inadequate with your suggestions! It's just that in this case, they'll be superseded by some existing, ongoing efforts around improving our promise rule docs to be more cohesive as a whole. 

Thanks for the PR and discussions! 🙂
